### PR TITLE
fix: ReactHooksRenderer renderHook was not expecting arguments

### PR DESCRIPTION
**Pull request by joshuaellis**
**What**:

ReactHooksRenderer was not expecting arguments due to `src/pure` missing those args in the `renderhook` function

**Why**:

fixes TS errors

**How**:

**Checklist**:

- [x] Documentation updated
- [x] Tests
- [x] Ready to be merged
      
- [x] Added myself to contributors table

### CODE REVIEW

**Comment by joshuaellis** [comment]
I tried this in my own project and got lots or warnings about this.

File: src/types/index.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { RenderHookOptions } from './internal'
```
Comment (mpeyper) [inline src/types/index.ts:1]: I'd probably move this out of `internal` now

File: src/types/index.ts [hunk 2]
```diff
@@ -48,7 +50,10 @@ export type RenderHook<
   AsyncUtils
 
 export interface ReactHooksRenderer {
-  renderHook: <TProps, TResult>() => RenderHook<TProps, TResult>
+  renderHook: <TProps, TResult, TOptions>(
+    callback: (props: TProps) => TResult,
+    options?: RenderHookOptions<TProps, TOptions>
+  ) => RenderHook<TProps, TResult>
```
Comment (mpeyper) [inline src/types/index.ts:56]: yep, I can see why you might be getting warnings about this :P

**Review by mpeyper — APPROVED** [review]

File: src/types/index.ts (hunk 1 above)
Comment (joshuaellis) [reply src/types/index.ts:1]: I did wonder that, wanted to get your confirmation. i'll move it then merge it

**Pull request merged**